src/shared_modules/utils/rsaHelper.hpp

```diff
@@ -92,16 +92,18 @@ class RSAHelper final
         // Defered free
         DEFER([&]() { T::RSA_free(rsa); });
 
-        // Decrypt the ciphertext using RSA private key
-        const auto decryptedLen = T::RSA_private_decrypt(RSA_SIZE,
+        // PKCS  #1  v1.5  padding.  This currently is the most widely used mode.  However, it is highly recommended to
+        // use RSA_PKCS1_OAEP_PADDING in new applications.
+        const auto decryptedLen = T::RSA_private_decrypt(input.length(),
                                                          reinterpret_cast<const unsigned char*>(input.data()),
                                                          reinterpret_cast<unsigned char*>(decryptedText.data()),
                                                          rsa,
-                                                         RSA_PKCS1_PADDING);
+                                                         RSA_PKCS1_OAEP_PADDING);
```

Comment:
This change breaks the compatibility with values already saved on the Keystore

The padding change is a recommendation from the library devs ([see documentation](https://www.openssl.org/docs/man1.1.1/man3/RSA_public_encrypt.html)):
>[!IMPORTANT]
> Decryption failures in the RSA_PKCS1_PADDING mode leak information which can potentially be used to mount a Bleichenbacher padding oracle attack. This is an inherent weakness in the PKCS #1 v1.5 padding design. Prefer RSA_PKCS1_OAEP_PADDING.

@Dwordcito I've performed some tests. If we maintain the old padding, the changes in this PR will be compatible with previous versions. Changing the padding breaks that compatibility but improves the security. How should we proceed?

Reply:
Maintain the old padding so as not to break upgrades.

Reply:
@MiguelazoDS Lets revert the padding changes, we are going to address this padding problem in another release (to not break retro-compatibility)

Great finding btw!